modelkit/core/profilers/base.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from typing import Generator, Dict, Set
+import time
+
+class BaseProfiler(ABC):
+    """Base Profiler class to be inherited for custom profiler."""
+
+    def __init__(self, model) -> None:
+        self.model = model
+        self.main_model_name = self.model.configuration_key
+        self._build(self.model)
+
+    @abstractmethod
+    def start(self, model_name: str) -> None:
+        """Define how to start recording an action."""
+
+    @abstractmethod
+    def end(self, model_name: str) -> None:
+        """Define how to record the cost when an action is completed."""
+
+    def summary(self, *args, **kwargs) -> str:
```

Comment:
I am wondering about the wording here.

Why a profiler should implement a `summary` method ?

A profiler need to give a way to query the metrics.

I was looking at https://pypi.org/project/prometheus-client/ or https://docs.datadoghq.com/fr/developers/dogstatsd/?tab=python but they both handle the exposure of metrics automatically.

Probably we should let each profiler implementation the choice on how they expose the metrics. ie. do not define this method in baseclass.
What do you think ?